src/modules/flow/iio/nodes.c

```diff
@@ -79,6 +70,20 @@ struct iio_node_type {
     void (*reader_cb)(void *data, struct sol_iio_device *device);
 };
 
+static inline struct sol_iio_channel *
+iio_add_channel(double scale, double offset, const char *name, struct iio_device_config *base)
+{
+    struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
+
+    if (base->use_device_default_scale)
+        channel_config.scale = scale;
+
+    if (base->use_device_default_offset)
+        channel_config.scale = offset;
```

Comment:
Here you probably meant `channel_config.offset = offset`. I guess that explain @laykuanloon scale issue.